radixdlt-java-common/src/main/java/com/radixdlt/errors/RadixErrors.java

```diff
@@ -62,38 +63,97 @@
  * permissions under this License.
  */
 
-package com.radixdlt.identifiers;
+package com.radixdlt.errors;
 
 import com.radixdlt.utils.functional.Failure;
 
-public enum CommonErrors implements Failure {
-	UNKNOWN_ERROR("Unknown error of type {0} with message {1}"),
+/**
+ * Full list of internal error codes.
+ * <p>
+ * <b>WARNING: New errors should be added to the end of the list to preserve error codes.</b>
+ */
+public enum RadixErrors implements Failure {
+	GENERAL("General error (used for testing only)"),
+	UNKNOWN("Unknown error of type {0} with message {1}"),
 
-	AID_STRING_IS_NULL("AID string is 'null'"),
 	ASYNC_PROCESSING_ERROR("Async processing error {0}"),
+
+	ERROR_CALL_DATA("{0}"),
+	ERROR_CONSTRAINT_VIOLATION("{0}"),
+	ERROR_DEFAULT_SYSTEM_LOAN("{0}"),
+	ERROR_DELEGATION_NOT_ALLOWED("{0}"),
+	ERROR_INVALID_HASHED_KEY("{0}"),
+	ERROR_INVALID_PERMISSION("{0}"),
+	ERROR_INVALID_RESOURCE("{0}"),
+	ERROR_INVALID_VIRTUAL_SUBSTATE("{0}"),
+	ERROR_LOCAL_SUBSTATE_NOT_FOUND("{0}"),
+	ERROR_MINIMUM_STAKE("{0}"),
+	ERROR_MISMATCH("{0}"),
+	ERROR_MISSING_PROCEDURE("{0}"),
+	ERROR_MULTIPLE_FEE_RESERVE_DEPOSIT("Multiple fee reserve deposit"),
+	ERROR_NOT_AUTHORIZED("{0}"),
+	ERROR_NOT_A_RESOURCE("{0}"),
+	ERROR_NOT_ENOUGH_RESERVE("{0}"),
+	ERROR_NOT_ENOUGH_RESOURCES("{0}"),
+	ERROR_PROCEDURE("{0}"),
+	ERROR_RESERVED_SYMBOL("{0}"),
+	ERROR_RESERVE_NOT_EMPTY("{0}"),
+	ERROR_RESOURCE_ALLOCATION_AND_DESTRUCTION("{0}"),
+	ERROR_SIGNED_SYSTEM("{0}"),
+	ERROR_SUBSTATE_NOT_FOUND("{0}"),
+	ERROR_VIRTUAL_PARENT_STATE_DOES_NOT_EXIST("{0}"),
+	ERROR_VIRTUAL_SUBSTATE_ALREADY_DOWN("{0}"),
+
+	INTERRUPTED_OPERATION("Operation interrupted with InterruptedException. Details: {0} {1}"),
+
 	INVALID_ACCOUNT_ADDRESS("Invalid account address {0}"),
-	INVALID_HEX_STRING("The value {0} is not a correct hexadecimal string"),
-	INVALID_LENGTH("AID string has incorrect length {0}"),
+	INVALID_ACTION_DATA("Invalid action data {0}"),
+	INVALID_AID_LENGTH("AID string has incorrect length {0}"),
+	INVALID_AID_STRING("AID string is 'null'"),
+	INVALID_PAGE_SIZE("Size {0} must be > 0"),
 	INVALID_PUBLIC_KEY("Invalid public key {0}"),
-	INVALID_RADIX_ADDRESS("Invalid RadixAddress {0}"),
-	INVALID_UINT_VALUE("Invalid UInt256/UInt384 value {0}"),
+	INVALID_RESOURCE_ADDRESS("Invalid resource address {0}"),
+	INVALID_SIGNATURE_DER("Invalid signature DER {0}"),
+	INVALID_TX_ID("Invalid TX ID {0}"),
 	INVALID_VALIDATOR_ADDRESS("Invalid validator address {0}"),
-	OPERATION_INTERRUPTED("Operation interrupted with InterruptedException. Details: {0} {1}"),
+	INVALID_VALUE_OUT_OF_RANGE("Parameter {0} must be between {1} and {2}"),
+	INVALID_VALUE_TYPE("Invalid value type {0}"),
+
+	IO_ERROR("I/O Error: {0} {1}"),
```

Comment:
I don't think we should be defining errors which aren't related to the api here. Any internal server/io errors should collapse to some "Internal Server" error.

Reply:
Moved to dedicated enum